### ISSUE TITLE
Accessibility problems

### ISSUE THREAD

**Issue by rebekaburnett** [facebook/docusaurus#10634]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [ ] I have tried the `npm run clear` or `yarn clear` command.
- [ ] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [ ] I have read the console error message carefully (if applicable).

### Description

We are required to pass the WCAG AA accessibility standards, and have found some problems in the Docusaurus parts. I think it might be easy for you to fix, and would be very important to us.

1. The links that open in a new tab/window should have an aria-label attribute or a screen-reader only element explaining that it opens in a new tab/window. We get this on every link that goes to another website. Here is an example: `<a href="https://example.com/" target="_blank" rel="noopener noreferrer" class="footer__link-item" >Example page</a>` (see attachment 1)
2. There is a size error on the DocSearch-Button-Key. (see attachment 2)
3. The hash links (the invisible markers immediately following each heading) are throwing contrast errors. If you add an aria label of hidden, it should fix that. (see attachment 3)

Thank you so much!! 

### Steps to reproduce

You can see the errors 1&2 by scanning the Docusaurs.io site with accessible:  `https://accessibe.com/accessscan?website=https://docusaurus.io/`

You can see error #3 by installing the WAVE extension in Chrome and enabling it (click the little W button in the header) when you're on one of the Docusaurus pages which have headers (e.g., https://docusaurus.io/docs)

### Expected behavior

Expect to not get any of the errors in the scans.

### Actual behavior

<img width="1368" alt="3-wave-hash-link" src="https://github.com/user-attachments/assets/7d271c9e-d1c0-4b5a-b5dd-f31d0a771f7b">
<img width="931" alt="2-small-size" src="https://github.com/user-attachments/assets/77ff0cb9-a14d-4bdc-9d19-9e311710ece8">
<img width="933" alt="1_links-open-tabs" src="https://github.com/user-attachments/assets/e1dd21c8-846b-4a51-8f8b-bcfff5913354">

### Your environment

- Public source code:
- Public site URL: https://developer.vippsmobilepay.com/
- Docusaurus version used: 3.5.2
- Environment name and version (e.g. Chrome 89, Node.js 16.4): Chrome 130
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): Windows 10

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by rebekaburnett** [comment]
I tested it locally, and it seemed to work to add the `visibility `tag to the `hash-link` style.


```
:global(.hash-link) {
  opacity: 0;
  visibility: hidden;
  ...
}

:global(.hash-link:focus),
:global(*:hover > .hash-link) {
  opacity: 1;
  visibility: visible;
}
```

**Comment by rebekaburnett** [comment]
I found that it was easy to fix locally 🙈
I just overwrite the hash-link style by adding it to my own custom.css.
If we use `visibility` instead of `opacity` we don't get the false alarm.

```
.hash-link {
  /* opacity: 0; */
  visibility: hidden;
  padding-left: 0.5rem;
  transition: opacity var(--ifm-transition-fast);
  -webkit-user-select: none;
          user-select: none;
}

.hash-link::before {
  content: '#';
}

.hash-link:focus,
*:hover > .hash-link {
  /* opacity: 1; */
  visibility: visible;
}
```
